pkg/collector/corechecks/containers/kubelet/provider/probe/provider.go

```diff
@@ -0,0 +1,110 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubelet
+
+package probe
+
+import (
+	"github.com/prometheus/common/model"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/kubelet/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/kubelet/provider/prometheus"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/tagger/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+// Provider provides the metrics related to data collected from the `/metrics/probes` Kubelet endpoint
+type Provider struct {
+	filter *containers.Filter
+	store  workloadmeta.Store
+	*prometheus.Provider
```

Comment:
can you explain why we use a pointer here for this composition?

Reply:
No reason, a remnant from when I was moving things around in my branch. The functionality is the same without it, so I will remove it

Reply:
I have removed it, thanks for pointing it out to me